File: src/Algorithm/Challenge/Brute-Forc/[leetcode.167]Two Sum II - Input array is sorted/TwoSum.js
// ideas: 暴力法—穷举查找
class Solution {
    constructor(nums, target) {
        this.nums = nums;
        this.target = target;
        this.lo = 0;
        this.hi = 0;
    }
    get lastIndex() {
        return this.nums.length - 1;
    }
    get result() {
        return [this.lo + 1, this.hi + 1]
    }
    isEqTarget() {
        let sum = this.nums[this.lo] + this.nums[this.hi];
        return sum == this.target;
    }
    twoSum() {
        while (this.lo < this.nums.length) {
            this.hi = this.lastIndex
            while (this.hi > this.lo) {
                if (this.isEqTarget()) return this.result;
                this.hi--
            }
            this.lo++
        }
    }
}

/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
    return new Solution(numbers, target).twoSum()
};

File: src/Algorithm/Challenge/leetcode.1/two_sum.c
// 求两数之和
#include <stdlib.h>

int* createResult(int a, int b){
    int* result = (int*)malloc(sizeof(int)*2);
    result[0] = a;
    result[1] = b;
    return result;
}

// twoSum
// 暴力法
int* twoSum(int* nums, int numsSize, int target, int* returnSize){
    int hi = numsSize - 1;

    for(int i = hi; i >= 0; i--){
        for(int j = 0; j < i; j++){
            if (nums[i] + nums[j] == target){
                *returnSize = 2;
                return createResult(j, i);
            }
        }
    }
    return 0;
}

